[WIP] 12.0 refactor

sebastienbeau opened the pull request:
Start to refactor
- [x] finish UI (partialy done)
- [x] simplify export
- [x] review key is only usable on basic field nether on relation
- [x] simplify import

next refactor will be done by  @kevinkhao
- [x] move test data into demo (in order to have some example)
- [x] replace "/key" by IDENTIFIER_SURFIXE
- [x] move to openpyxl fix xls constraint (https://openpyxl.readthedocs.io/en/stable/validation.html)
- [x] rework xls part
- [x] rework the import part (extract do too much thing and error are raise in the job)
- [x] make all test green

sebastienbeau committed:
[REF] refactor code

- improve UI in order to force the user to fill all field needed
- reimplement constraint and only apply them on "pattern" export
- add test

sebastienbeau committed:
[REF] simplify code for building header

sebastienbeau committed:
[REF] refactor export, use base_jsonify to simplify the code

sebastienbeau committed:
[REF] reimplement is key in export and improve UI

sebastienbeau committed:
[REF] reimplement is key in export and improve UI

sebastienbeau committed:
[IMP] finish UI for select tab

sebastienbeau commented on pattern_import_export/tests/test_pattern_export.py:
```diff
@@ -38,19 +36,18 @@ def test_get_header3(self):
         @return:
         """
         export_fields_m2m = self.ir_exports_m2m.export_fields.filtered(
```
@kevinkhao it will be simplier to have demo data for profile and here update the profile line with the xmlid

sebastienbeau committed:
[REF] refactor import and based in on odoo standard import

sebastienbeau committed:
[IMP] improve import and add xlsx test and return error in file

sebastienbeau committed:
[FIX] Remove useless constraint as we now can have several key

kevinkhao committed:
[IMP] Moved some record creation in tests to demo.xml (#1)

* [IMP] Moved some record creation in tests to demo.xml

* [IMP] More cleanup on tests

* [IMP] Removed superfluous objects in common tests

* [IMP] Cleared logic on _get_select_tab, added test for it

* [IMP] Misc. improvements according to feedback

* [FIX] Linter problems

* [IMP] Cleanup

* [FIX] Revert bad changes, misc imp

kevinkhao commented:
Hi @sebastienbeau 
* [x]  replace "/key" by IDENTIFIER_SURFIXE - done
* [x]  move to openpyxl fix xls constraint (https://openpyxl.readthedocs.io/en/stable/validation.html) - done
* [x]  rework xls part - done, and base module has been modified. Some things that are changed: tabs don't exist, filters are used instead, some methods moved from export to export line
* [ ]  make all test green - imports tests are broken, I haven't touched them yet. other base module + XLSX module tests are OK


Those are done, some TODO's are left.
The PR is here
https://github.com/akretion/pattern-import-export/pull/2
It might break some stuff you were working on because of some changes. 
I have done technical testing mostly. I haven't thoroughly tested it on functional side.

sebastienbeau committed:
Merge pull request #4 from akretion/12.0-import-refactor

12.0 import refactor

kevinkhao committed:
[FIX] Make tests green

kevinkhao committed:
[IMP] Change constant /key to #key, move constants to common.py

kevinkhao committed:
[FIX] Cleanup to make Travis happy

kevinkhao committed:
[FIX] Cleanup to make Travis happy

kevinkhao committed:
Merge xlsx refactor

sebastienbeau committed:
Merge pull request #6 from akretion/12.0-refactor-fixtests

Fix test, small cleanup to make travis green

sebastienbeau committed:
Merge pull request #7 from akretion/12.0-refactor-key-separator

Update key separator, move key separator and column separator to common

kevinkhao committed:
[TMP] swap

kevinkhao committed:
[FIX] Fix tests after moving to openpyxl

kevinkhao committed:
[WIP] Add specific model and menu for patterned import/export jobs

kevinkhao committed:
[WIP] Adding excel tests to cover failure with bad id values

kevinkhao committed:
[WIP] Adding excel tests to cover failure with bad id values

kevinkhao committed:
Add some tests for bad data

kevinkhao committed:
Add some tests for bad data

kevinkhao committed:
Modify tests to pass before merging

kevinkhao committed:
Modify tests to pass before merging

kevinkhao committed:
Update pattern_import_export_xlsx/tests/test_pattern_import.py

Co-authored-by: beau sebastien <sebastien.beau@akretion.com>

kevinkhao committed:
Update pattern_import_export_xlsx/tests/test_pattern_import.py

Co-authored-by: beau sebastien <sebastien.beau@akretion.com>

kevinkhao committed:
[IMP] Modifications according to feedback
- Change "use_tab" to "tab_filter_id"
- Add domain on UI on tab_filter_id
- Small optimization on _populate_main_sheet_rows
- Remove some unused code

kevinkhao committed:
[IMP] Modifications according to feedback
- Change "use_tab" to "tab_filter_id"
- Add domain on UI on tab_filter_id
- Small optimization on _populate_main_sheet_rows
- Remove some unused code

sebastienbeau committed:
Merge pull request #8 from akretion/12.0-refactor-more-tests

More failing tests

kevinkhao committed:
[IMP] Notification work
- Add methods to provide message content for notifications
- Add notification once job is finished to inform user of import/export state
- Add URL to notification for direct access to binary content (export) or form view for more info (import)
- Reorganized some code to keep+display job result in a simple way

chafique-delli committed:
[FIX] db_id_for and _str_to_boolean methods

sebastienbeau committed:
[IMP] allow to have empty line in teh file and commented columns starting by #

sebastienbeau committed:
[FIX] depending of the original source of thx xlsx the function =TRUE() and =FALSE() are valid boolean

sebastienbeau committed:
[IMP] allow to import a specif tab in a file

sebastienbeau committed:
[FIX] add support of global message

sebastienbeau committed:
[IMP] improve UI add the possibility to import from the pattern

sebastienbeau committed:
Merge branch '12.0-refactor' into 12.0-refactor-metadata-notif

sebastienbeau committed:
[FIX] fix import of empty m2m

sebastienbeau committed:
Merge pull request #10 from akretion/12.0-refactor-metadata-notif

Add user notification when import/export job finished with debugging info

sebastienbeau committed:
[FIX] fix issue when importing empty o2m column

sebastienbeau committed:
[FIX] fix writting in Error column

sebastienbeau committed:
[FIX] fix issue when record are missing

sebastienbeau committed:
[IMP] add test for testing import with key

sebastienbeau committed:
[FIX] fix converting boolean and fix reading file

sebastienbeau committed:
[FIX] add default value for tab_to_export

sebastienbeau committed:
Merge pull request #12 from akretion/12.0-UX-behaviour-improvement

12.0 ux behaviour improvement

sebastienbeau committed:
[FIX] fix raising error

sebastienbeau committed:
[IMP] take care of existing domain on the field

sebastienbeau committed:
[FIX] solve issue with empty item

sebastienbeau committed:
[FIX] solve issue import with empty id or external id

sebastienbeau committed:
[IMP] support many2one for identifier key

sebastienbeau committed:
[IMP] support many2one for identifier key

sebastienbeau committed:
[FIX] fix pre-commit

sebastienbeau committed:
[IMP] improve view allow to download directly the attachment

sebastienbeau committed:
Merge pull request #13 from akretion/12.0-refactor-import-key

12.0 refactor import key

sebastienbeau commented:
TODO https://github.com/akretion/pattern-import-export/pull/14
- [x] fix import of record with parent (like category)
- [x] manage multi header at import
- [x] rename state when creating a patterned import

kevinkhao committed:
[IMP] Add possibility of having non-technical descriptive header in addition to technical name header in imports and imports
Set default state of patterned import export as "pending" instead of "fail"

kevinkhao committed:
[IMP] Add functionality and tests for supporting m2o parent-child relationships

sebastienbeau committed:
[FIX] fix m2m import when importing with standard odoo

sebastienbeau committed:
Merge pull request #15 from akretion/12.0-fix-m2m-standard-import

[FIX] fix m2m import when importing with standard odoo

sebastienbeau committed:
[FIX] fix clear error col

sebastienbeau committed:
[REF] refactor code to avoid to inherit load method

sebastienbeau commented:
TODO @kevinkhao 
- [ ] improve UI split import/export menu in two
- [ ] import patterned split result in two field (too avoid a long list of ids in the tree view)
![image](https://user-images.githubusercontent.com/1164578/93816618-0f2ab480-fc58-11ea-8537-07e8f182d0ca.png)

- [ ] add smart button on ir.exports menu (with number of done/fail/pending)
- [ ] add missing tab_to_import in view
- [ ] in the tree/form view of partterned import add the link of the pattern

kevinkhao committed:
[IMP] Remove useless self.env.clear() in tests

kevinkhao committed:
[IMP] Merge duplicate with_context()

kevinkhao committed:
[IMP] Separate patterned imports and exports into their own menu
Separate info field into info and info_separate for better tree view experience

kevinkhao committed:
[IMP] Add export_id to patterned.import.export object for ease of finding our way back to the pattern that was used

sebastienbeau committed:
Merge branch '12.0-refactor-multiheader-misc' into 12.0-refactor

kevinkhao committed:
[IMP] Add smart button to open patterned exports/imports from the ir.exports form view, and its associated record counts
Remove the ability to create a patterned import/export from tree or form view

kevinkhao commented:
https://github.com/akretion/pattern-import-export/pull/16

sebastienbeau committed:
[REF] better naming

kevinkhao committed:
[FIX] Trivial formatting commit to avoid pre-commit complaining: it doesn't know what it wants because it applies black which extends a line then flake8 complains about length

sebastienbeau committed:
[IMP] add partial commit

When importing a huge file that take a long time it's quiet borring to
need to reimport all valid line
Here we add the possibility to commit every valid line, then you can do
a smaller file with only failling line
Add also the possibility to configure the flush step.
Indeed when doing an import of a huge file if one line fail then odoo
will import the line one by one and it will so fucking long
doing smaller flush reduce this issue

sebastienbeau committed:
Merge pull request #17 from akretion/12.0-partial-commit

12.0 partial commit

sebastienbeau committed:
[IMP] improve UI

sebastienbeau committed:
Merge pull request #21 from akretion/12.0-refactor-misc-ui-improvements

12.0 refactor misc ui improvements

sebastienbeau commented:
It's time to merge this refactor ! /ocabot merge major
